articles/understanding-tcp-ip-transport-layer-protocols/index.md

```diff
@@ -0,0 +1,164 @@
+Assuming in the example above, the host received the first hundred bytes (0-99) with a sequence number of 1, and instead of receiving the next bytes (100-199) with a sequence number of 2, the host receives a segment with (200-299) bytes with a sequence number of 3, it will send a packet requesting for that missing segment, i.e., 100-99 with a sequence number of 2. The sending device can also resend a segment if the receiving device does not acknowledge all the data sent. The sending device, in this case, is a server.  
+The server waits for a few moments using a timer called the retransmission timer to make sure that no other acknowledgment arrives, after which it then decides to resend that particular segment that it did not receive its acknowledgment flag.  
+
+**Flow control using Widowing**  
+Because network host has limited resources such as limited space and processing power, TCP implements a mechanism called flow control using a window concept that applied to the amount of data that can be awaiting acknowledgment at any one point in time.
+The receiving device uses the windowing concept to inform the sender how much data it can receive at any given time; this allows the sender to either speed up or slow down the sending of segments through a **window sliding process.**
+
+### User Datagram Protocol
+
+User datagram protocol (UDP) is considered as best-effort transport protocol because it is a light-weighted transport protocol. UDP is a connectionless protocol, meaning it provides no reliability nor reordering of the data segment and flow control like TCP, which provides these services. Because of this, UDP is faster than TCP in transporting data. However, according to [Cisco](https://www.briefmenow.org/cisco/what-will-happen-17), UDP provides some similar benefits to TCP, such as data segmentation and Multiplexing using port numbers. UDP is used by applications that are tolerant to loss of data but not delay. For example, TCP's requirement will make it difficult to stream live video as all packets must be sent and acknowledged, which will consume many resources and causes severe delay. 
+ But with UDP, if a packet is missing, the streaming will continue unnoticed. It only becomes apparent when many segments are missing, which is seen in low video quality and lack of synchronization between video and audio. 
+The significant difference between TCP and UDP is that TCP offers a wide range of services to applications, while UDP does not, this does not make UDP inferior to TCP, but by providing fewer services, UDP has fewer bytes in its header, and this makes UDP is faster in transporting data.
+![UDP header](/engineering-education/understanding-tcp-ip-transport-layer-protocols/udp-image.png)
+![Source](https://en.wikiversity.org/wiki/Internet_Protocol_Analysis/Transport_Layer#/media/File:Internet_Protocol_Analysis_-_Transport_Layer.png)
```

Comment:
Remove `!`

Reply:
@raboba2re
Just a minor thing to note.
In markdown, for rendering images you have to add an `!`, and for sources you need not. In the latest fixes, I saw you had removed `!` from both the places, which is incorrect.

Have a look at my commits to understand.